src/utils/utils/units.ts

```diff
@@ -63,3 +49,90 @@ export const getHumanizedSize = (size: string, bytesOption: 'withB' | 'withoutB'
 
   return humanizeBinaryBytes(baseValue);
 };
+
+const extractUnitFromQuantityString = (quantityString: string) =>
+  quantityString.match(UNIT_REGEX)?.[0];
+
+const extractNumberFromQuantityString = (quantityString: string) => {
+  const match = quantityString.match(NUMBER_REGEX);
+  return match ? parseFloat(match[0]) : null;
+};
+
+export const isBinaryUnit = (unit: string): unit is BinaryUnit =>
+  binaryUnitsOrdered.includes(BinaryUnit[unit]);
+
+export const isDecimalUnit = (unit: string): unit is DecimalUnit =>
+  decimalUnitsOrdered.includes(DecimalUnit[unit]);
+
+const isQuantityUnit = (unit: string): unit is QuantityUnit =>
+  isBinaryUnit(unit) || isDecimalUnit(unit);
+
+export const toQuantity = (quantityString: string, keepInitialUnit = true): Quantity => {
+  const preferredUnit = keepInitialUnit ? extractUnitFromQuantityString(quantityString) : null;
+
+  if (isQuantityUnit(preferredUnit)) {
+    return {
+      unit: preferredUnit,
+      value: extractNumberFromQuantityString(quantityString),
+    };
+  }
+
+  // convert bytes or millibytes to a larger unit (if the corresponding value for that unit is >= 1)
+  // value CAN BE a floating point number
+  const { unit, value } = getHumanizedSize(quantityString, 'withoutB');
+
+  return { unit, value };
+};
+
+export const quantityToString = ({ unit, value }: Quantity) =>
+  `${value}${unit === BinaryUnit.B ? '' : unit}`;
+
+const convertBytes = (value: number, base: 'BINARY' | 'DECIMAL') => {
+  const divisor = base === 'BINARY' ? 1024 : 1000;
+  let index = 0;
+
+  while (value % divisor === 0) {
+    value /= divisor;
+    index++;
+  }
+
+  return { unitIndex: index, value };
+};
+
+const getFormattedString = (value: number, unit: QuantityUnit) =>
+  `${value}${unit === 'B' ? '' : unit}`;
+
+/**
+ * Formats the quantity string that is in bytes format to a more simplified quantity string in integer + unit format (if the conversion to an integer is possible).
+ * e.g. '1024' to 1Ki or '1000000' to 1M
+ *
+ * @param quantityString string to format, must be in a Kubernetes Quantity format: https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/
+ * @returns formatted quantity string
+ */
+export const formatQuantityString = (quantityString: string) => {
+  if (!quantityString) {
+    return null;
+  }
+
+  const unit = extractUnitFromQuantityString(quantityString);
+
+  // special case when unit = 'm' can occur, e.g. if we create a PVC with floating point number like 0.2Gi (1/5 Gi) which can't be converted to bytes (power of 2 is not divisible by 5), 'm' then stands for milibytes
+  if (isQuantityUnit(unit) || unit === 'm') {
```

Comment:
it seems we have more cases like this - https://github.com/kubevirt/kubevirt/blob/205d9455db56d5fcd42fb331122cfef358f19a69/vendor/k8s.io/apimachinery/pkg/api/resource/suffix.go#L123